### ISSUE TITLE
Support for Span link with good DX interface and demo presentation

### ISSUE THREAD

**Issue by ManiMozaffar** [pydantic/logfire#158]
### Description

I couldn't find anywhere in docs where it mentioned about span links. That might be something logfire is missing at the moment.
https://opentelemetry.io/docs/concepts/signals/traces/#span-links

It'd be nice to also have this on example, linking a span from frontend, to the backend. I'm interested how that would look in the dashboard as well.
Currently otel sdk interface for span link is some kind of horrible. but I'm questioning if that can be designed better or not.

**Comment by adriangb** [comment]
It's something we're actively working on adding.

Do you have any suggestions for APIs you'd like in the SDK?

**Comment by ManiMozaffar** [comment]
> Do you have any suggestions for APIs you'd like in the SDK?

Not really, we instrumented kafka batch process and consuming, which then show a really nice links from the first span (message) to the last one. but overall the module that instrumented were messy a bit and highly complicated.
But what I find it even more complicated, is how to allow react, our client, to link spans from frontend to the related backend request? I found this kinda complicated, that's why if this is show-cased in the demo project, it'd be awesome.

**Comment by Kludex** [comment]
We already have support to span links via `logfire.span(..., _links=...)`, but I think we can improve our API.

I've implemented https://github.com/pydantic/logfire/pull/726 to reopen the discussion.

**Comment by dmontagu** [comment]
Closing — span links are supported via `logfire.span(..., _links=...)`. If there's a specific DX improvement you'd still like, a fresh issue with the concrete API would be easier to act on.
